src/java.base/share/classes/java/util/concurrent/CopyOnWriteArrayList.java

```diff
@@ -328,7 +335,7 @@ public Object clone() {
      * @return an array containing all the elements in this list
      */
     public Object[] toArray() {
-        return getArray().clone();
+        return getArray().length == 0 ? getArray() : getArray().clone();
```

Comment:
Unfortunately, I think this is against the spec for this method, which explicitly says the method must allocate a new array. Yes, this change would be within the spirit of the spec ("You can modify the array", which you cannot if the object is empty), but is against the letter of it.

Reply:
I think you're right, I'll remove the change to toArray().